inference-engine/src/multi_device/multi_device_plugin.cpp

```diff
@@ -137,11 +142,28 @@ std::vector<DeviceInformation> MultiDeviceInferencePlugin::ParseMetaDevices(cons
 
         std::string defaultDeviceID = "";
         DeviceIDParser parsed{deviceName};
-        if (parsed.getDeviceID().empty())
+        std::string deviceid = parsed.getDeviceID();
+        if (deviceid.empty()) {
             defaultDeviceID = getDefaultDeviceID(deviceName);
+            deviceid = defaultDeviceID;
+        }
 
+        std::string fullDeviceName = "";
+        std::string uniqueName = "";
+        std::vector<std::string> supportedMetrics = GetCore()->GetMetric(parsed.getDeviceName(), METRIC_KEY(SUPPORTED_METRICS));
+        if (std::find(supportedMetrics.begin(), supportedMetrics.end(), METRIC_KEY(FULL_DEVICE_NAME)) != supportedMetrics.end()) {
+            fullDeviceName = GetCore()->GetMetric(parsed.getDeviceName(), METRIC_KEY(FULL_DEVICE_NAME)).as<std::string>();
```

Comment:
@myshevts , so when I get FULL_DEVICE_NAME , I should use deviceName or parsed.getDeviceName() here

Reply:
test on machine  with myriad, their device name is deviceNameWithID, but on machine with GPU, the IGPU and dGPU deviceName is all GPU, so only GPU need the full device Name for the unique name